src/Kernel/client/plotting.ts

```diff
@@ -1,12 +1,79 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT License.
 
-function importScript(url: string) {
-    let script = document.createElement("script");
-    script.setAttribute("src", url);
-    document.body.appendChild(script);
+// This is a bit of a hack needed to map the requireJS
+// call made by TypeScript onto the URL that Jupyter
+// makes our kernelspec available at.
+//
+// Using this hack, we can split the type import and
+// the runtime import apart, then glue them back
+// together using a declare global to solve TS2686.
+/// <amd-dependency path="/kernelspecs/iqsharp/chart.js" name="Chart" />
+import type * as ChartJs from "chart.js";
+declare global {
+    const Chart: typeof ChartJs;
 }
 
+export interface Complex {
+    Real: number;
+    Imag: number;
+    Magnitude: number;
+    Phase: number;
+};
+
+export interface DisplayableState {
+    n_qubits: number;
+    div_id: string;
+    amplitudes: Complex[] | null;
+};
+
+
 export function initializePlotting() {
-    importScript("https://cdn.jsdelivr.net/npm/chart.js@2.8.0");
-}
+    // importScript("https://cdn.jsdelivr.net/npm/chart.js@2.8.0");
+};
+
+
+
+export function createBarChart(element: HTMLCanvasElement, state: DisplayableState) {
+    var amps = state.amplitudes;
+
+    let newCount = amps.length;
+    let nQubits = Math.log2(newCount) >>> 0;
```

Comment:
Similarly here, this should be replaced by looking at `state.n_qubits`.

Reply:
got rid of the two variables and just used state.n_qubits